### PR TITLE
Add a Reactive MySQL client extension

### PR DESCRIPTION
**Pull request by tsegismont**
New extension similar to the Postgres extension (no shared code yet).
Integrations tests with MariaDB.

### CODE REVIEW

**Comment by tsegismont** [comment]
@cescoffier @FroMage can you please take a first look?

This PR is based on #3711 

We need to agree on a documentation update. Here's my proposal:

- we rename the _Reactive Postgres Client_ guide to _Reactive DB Clients_
- we add some text to tell users:
  - which databases are supported
  - examples use Postgres
  - how to configure Postgres
  - how to configure MySQL

**Review by FroMage — APPROVED** [review]
LGTM

**Comment by tsegismont** [comment]
@FroMage any comment about the doc issue mentioned above?

**Comment by geoand** [comment]
Vert.x is being updated to `3.8.1` as part of #3746

**Comment by FroMage** [comment]
> @FroMage any comment about the doc issue mentioned above?

Oh, about the guide update? Yeah, your plan looks good: no need to write two guides at this point: there's not enough differences in the two clients to warrant it.

**Comment by tsegismont** [comment]
@FroMage just update the doc

File: docs/src/main/asciidoc/reactive-database-clients.adoc [hunk 1]
```diff
@@ -3,14 +3,20 @@ This guide is maintained in the main Quarkus repository
 and pull requests should be submitted there:
 https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
 ////
-= Quarkus - Reactive Postgres Client
+= Quarkus - Reactive Database Clients
```
Comment (BillyYccc) [inline docs/src/main/asciidoc/reactive-database-clients.adoc]: How about naming it `the Reactive SQL Client`, just to be the same as what the original one is.

**Comment by FroMage** [comment]
CI failed in devmojo (unrelated), restarting them.

**Comment by geoand** [comment]
Restarted again

**Comment by FroMage** [comment]
Now failed in kogito. Again doesn't seem related.

**Comment by FroMage** [comment]
@tsegismont is it still not ready for merge? Can you please rebase just to check if it fixes CI?

**Comment by FroMage** [comment]
Thanks, crossing CI fingers.

File: docs/src/main/asciidoc/reactive-database-clients.adoc (hunk 1 above)
Comment (tsegismont) [reply docs/src/main/asciidoc/reactive-database-clients.adoc]: Good point @BillyYccc

Comment (tsegismont) [reply docs/src/main/asciidoc/reactive-database-clients.adoc]: Done

**Pull request merged**